test/test_python_dispatch.py

```diff
@@ -1324,6 +1330,52 @@ def f(x, s, i):
         self.assertEqual(len(counters['graph_break']), 0)
 
 
+class TestCompileCheck(TestCaseWithCustomOpTearDown):
+    @ops(custom_op_db, dtypes=OpDTypes.any_one)
+    def test_compile_check_op(self, device, dtype, op):
+        for sample_input in op.sample_inputs(device, dtype, requires_grad=op.supports_autograd):
+            dynamic_only = op.name in ("NumpyNMSCustomOp", "NumpyNonzeroCustomOp")
+            args = [sample_input.input] + list(sample_input.args)
+            kwargs = sample_input.kwargs
+            inference_only = not op.supports_autograd
+            compile_check(
+                op.op, args, kwargs,
+                inference_only=inference_only,
+                dynamic_only=dynamic_only,
+                fullgraph=False,  # Dynamo graph breaks on CustomOp today
+            )
+
+    def test_compile_check_fails_basic(self, device):
```

Comment:
I see there are some basic tests for failure cases. Since the main use case for this API is giving users quick diagnostics on whether their custom ops are broken (and how to fix them), is it worth adding a few tests for some of the failure modes we know of? We could also expecttest assert that the error message is reasonable, although maybe that's overkill.

- missing fake impl
- doesn't work with dynamic shapes
- schema aliasing annotations are wrong
- forward impl directly returns one of the input tensors
- autograd formula is wrong (gradcheck error)

Reply:
More tests are worth it, and ideally they would assert that the error message is reasonable.